kvbc/include/v4blockchain/detail/keys_history.h

```diff
@@ -0,0 +1,124 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "rocksdb/native_client.h"
+#include "categorization/updates.h"
+#include "v4blockchain/detail/categories.h"
+#include <rocksdb/compaction_filter.h>
+
+/*
+ * TODO / questions / notes (efrat):
+ * - all values in CF are empty string
+ * - make MAX_BLOCKS_NUM_SAVE_KEYS configurable
+ *
+ * */
+
+namespace concord::kvbc::v4blockchain::detail {
+
+class KeysHistory {
+ public:
+  KeysHistory(const std::shared_ptr<concord::storage::rocksdb::NativeClient>&,
+              const std::optional<std::map<std::string, concord::kvbc::categorization::CATEGORY_TYPE>>&);
+  void addBlockKeys(const concord::kvbc::categorization::Updates&, const BlockId, storage::rocksdb::NativeWriteBatch&);
+
+  // get the value and return deserialized value if needed.
+  std::optional<BlockId> getUpdateVersionFromSnapshot(const std::string& category_id,
```

Comment:
Let's rename this method and not use snapshot in the key history class, as it's confusing.

Reply:
ok, changed to getVersionFromHistory, is that a clear name in your opinion?
maybe getUpdateVersionFromHistory or getChangeVersionFromHistory are better?